### DISCUSSION TITLE
Monte-Carlo, Worst-Case and sensitivity analysis

### DISCUSSION THREAD

**Discussion by efa** [ra3xdh/qucs_s#906] [Ideas]
hi,
Cadence PSpice can do Monte-Carlo, Worst-Case and sensitivity analysis.
Are those possible with Qucs-S/ngspice ?

**Comment by ra3xdh** [comment]
The Monte-Carlo simulation is only possible using Nutmeg scripting mode. Refer to the attached example. 
[MC_qucs.zip](https://github.com/user-attachments/files/16737440/MC_qucs.zip)

**Comment by ra3xdh** [comment]
The adding of automated MonteCarlo simulation could be considered in the future.

**Comment by efa** [comment]
in the ngspice docs cap.18:
https://ngspice.sourceforge.io/docs/ngspice-html-manual/manual.xhtml#magicparlabel-31540
https://ngspice.sourceforge.io/docs/ngspice-html-manual/manual.xhtml#magicparlabel-31719
I can see it can do Monte Carlo using unif() or gauss() for a parameter, or using its control language.

Knowing the mean electronic designer, a GUI to easily run Monte Carlo with number of run, limits and so, like in Cadence PSpice can be very good to attract users and increase the usefulness of Qucs-S
